src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -143,6 +156,14 @@ assign(ReactDOMTextComponent.prototype, {
   },
 
   unmountComponent: function() {
+    var openingComment = this._openingComment;
+    if (openingComment) {
+      DOMChildrenOperations.removeDelimitedText(
```

Comment:
What do you think of returning from getNativeNode an two-element array with the `[open, close]` comments? Then you would have both nodes already and wouldn't need to walk backwards in DOMChildrenOperations. (Let me know if this wouldn't work for some reason – what you have currently also seems reasonable, albeit a little clunky.)